crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -472,7 +471,7 @@ async fn parse_c8y_topics(
                 ..
             },
         ) => {
-            let topic = C8yTopic::SmartRestResponse.to_topic()?;
+            let topic = C8yTopic::SmartRestResponse(None).to_topic()?;
```

Comment:
When I see such a `SmartRestResponse(None)` I wonder what could have been this `None`. It's not obvious that this a child device id.

Why not something similar to:

```
pub enum C8yTopic {
    SmartRestRequest,
    SmartRestResponse,
    ChildSmartRestResponse(String),
    OperationTopic(String),
}
```

This also raises another question (that is beyond that PR):
* Why a response topic for the child devices, but nothing similar for the requests and the operations?

Reply:
I agree that having a new variant is better than overloading the existing one. Regarding the lack of "requests" and "operations" for child devices, it is as designed by Cumulocity. There's no `s/ds/{child-id}` topic as the requests for the parent device as well as child devices are received on `s/ds` topic with the `child-id` in the payload.

Although inconsistent with `s/us/{child-id}`, this design is beneficial to us as we don't have to keep restarting the msoquitto bridge to subscribe to new child topics whenever a new child gets added.